src/utils/walletConnect.ts

```diff
@@ -261,3 +321,169 @@ export function onSessionProposal(
+        });
+
+        return;
+      }
+
+      // for TS only, should never happen
+      if (!allWallets) {
+        logger.error(new RainbowError(`WC v2: allWallets is null`));
+        return;
+      }
+
+      const selectedWallet = findWalletWithAccount(allWallets, address);
+
+      if (!selectedWallet || selectedWallet?.type === WalletTypes.readOnly) {
+        logger.debug(
+          `WC v2: session_request exited, selectedWallet was falsy or read only`
+        );
+
+        await client.respond({
+          topic,
+          response: formatJsonRpcError(id, `Wallet is read-only`),
+        });
+
+        return;
+      }
+    } else {
+      address = params[0].from;
+    }
+
+    const session = client.session.get(topic);
+    const { nativeCurrency, network } = store.getState().settings;
+    const chainId = Number(event.params.chainId.split(':')[1]);
+    const dappNetwork = ethereumUtils.getNetworkFromChainId(chainId);
+    const displayDetails = getRequestDisplayDetails(
+      event.params.request,
+      nativeCurrency,
+      dappNetwork
+    );
+    const peerMeta = session.peer.metadata;
+    const request: RequestData = {
+      clientId: session.topic, // I don't think this is used
+      peerId: session.topic, // I don't think this is used
+      requestId: event.id,
+      dappName:
+        dappNameOverride(peerMeta.name) || peerMeta.name || 'Unknown Dapp',
+      dappScheme: 'unused in WC v2', // only used for deeplinks from WC v1
+      dappUrl: peerMeta.url || 'Unknown URL',
+      displayDetails,
+      imageUrl: maybeSignUri(
+        dappLogoOverride(peerMeta.url) || peerMeta.icons[0]
+      ),
+      payload: event.params.request,
+      walletConnectV2RequestValues: {
+        sessionRequestEvent: event,
+        // @ts-ignore we assign address above
+        address, // required by screen
+        chainId, // required by screen
+      },
+    };
+
+    logger.debug(`request`, { request });
+
+    const { requests: pendingRequests } = store.getState().requests;
+
+    if (!pendingRequests[request.requestId]) {
+      const updatedRequests = {
+        ...pendingRequests,
+        [request.requestId]: request,
+      };
+      store.dispatch({
+        payload: updatedRequests,
+        type: REQUESTS_UPDATE_REQUESTS_TO_APPROVE,
+      });
+      saveLocalRequests(updatedRequests, address, network);
+
+      logger.debug(`WC v2: navigating to CONFIRM_REQUEST sheet`);
+
+      Navigation.handleAction(Routes.CONFIRM_REQUEST, {
+        openAutomatically: true,
+        transactionDetails: request,
+      });
+
+      analytics.track('Showing Walletconnect signing request', {
+        dappName: request.dappName,
+        dappUrl: request.dappUrl,
+      });
+    }
+  } else {
+    logger.info(`utils/walletConnectV2: received unsupported session_request`);
+
+    await client.respond({
+      topic,
+      response: formatJsonRpcError(id, `Unsupported RPC method`),
+    });
```

Comment:
We used to proxy these on to our RPC endpoints. Need to double check if this is still necessary.